packages/web3-validator/src/web3_validator.ts

```diff
@@ -15,59 +15,29 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-import Ajv from 'ajv';
-import { blake2b } from 'ethereum-cryptography/blake2b';
-import { utf8ToBytes, toHex } from 'ethereum-cryptography/utils';
+import { Validator } from './validator';
 import { ethAbiToJsonSchema } from './utils';
-import { ValidationSchemaInput, Web3ValidationErrorObject, Web3ValidationOptions } from './types';
-import { ethKeyword } from './keywords/eth';
+import {
+	Json,
+	ValidationSchemaInput,
+	Web3ValidationErrorObject,
+	Web3ValidationOptions,
+} from './types';
 import { Web3ValidatorError } from './errors';
-import * as formats from './formats';
 
 export class Web3Validator {
-	private readonly _validator: Ajv;
+	private readonly _validator: Validator;
 
 	public constructor() {
-		this._validator = new Ajv({
-			strict: true,
-			strictSchema: true,
-			allErrors: true,
-			useDefaults: false,
-			// To avoid warnings for not defining `type` for each property
-			strictTypes: false,
-
-			// To avoid strict mode error for minItems or maxItems/additionalItems
-			strictTuples: false,
-		});
-
-		this._validator.addKeyword(ethKeyword);
-
-		for (const formatName of Object.keys(formats)) {
-			// eslint-disable-next-line import/namespace
-			this._validator.addFormat(formatName, formats[formatName as keyof typeof formats]);
-		}
+		this._validator = new Validator();
```

Comment:
should we have singleton Validator so we ensure that only one entry is made into its map for each schema.

Reply:
and that there is only one instance of validator and one map across lib for managing schemas.

Reply:
I've committed a singleton